File: codeforces/contest/1959/C/main.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    int T;
    cin >> T;
    while (T--) {
        int N, K;
        cin >> N >> K;
        vector<int> lC(26), uC(26);
        while (N--) {
            char c;
            cin >> c;
            if (islower(c)) {
                lC[c - 'a']++;
            } else {
                uC[c - 'A']++;
            }
        }
        int res = 0, diff_sum = 0;
        for (int i = 0; i < 26; i++) {
            res += min(lC[i], uC[i]);
            diff_sum += abs(lC[i] - uC[i]) / 2;
        }
        cout << res + min(diff_sum, K) * 2 << "\n";
    }
}

File: codeforces/contest/1959/C/main.kt
import kotlin.math.abs
import kotlin.math.min

fun main() {
    val T = readLine()!!.toInt()
    repeat(T) {
        val (N, K) = readLine()!!.split(" ").map { it.toInt() }
        val lC = IntArray(26)
        val uC = IntArray(26)
        readLine()!!.forEach { c ->
            if (c.isLowerCase()) {
                lC[c - 'a']++
            } else {
                uC[c - 'A']++
            }
        }
        var res = 0
        var diffSum = 0
        for (i in 0 until 26) {
            res += min(lC[i], uC[i])
            diffSum += abs(lC[i] - uC[i]) / 2
        }
        println(res + min(diffSum, K))
    }
}
